docs/debugging-transactions.md

```diff
@@ -0,0 +1,312 @@
+            "output": "0x000000000000000000000000000000000000000000000000000000000001371e",
+            "calls": [
+               {
+                  "type": "STATICCALL",
+                  "from": "0x3ee18b2214aff97000d974cf647e7c347e8fa585",
+                  "to": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
+                  "gas": "0x123bb",
+                  "gasUsed": "0x25c0",
+                  "input": "0x313ce567",
+                  "output": "0x0000000000000000000000000000000000000000000000000000000000000006",
+                  "calls": [
+                     {
+                        "type": "DELEGATECALL",
+                        "from": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
+                        "to": "0xa2327a938febf5fec13bacfb16ae10ecbc4cbdcf",
+                        "gas": "0x10357",
+                        "gasUsed": "0x94d",
+                        "input": "0x313ce567",
+                        "output": "0x0000000000000000000000000000000000000000000000000000000000000006"
+                     }
+                  ]
+               }
+            ]
+         }
+      ]
+   }
+}
+```
+
+#### 2. With `opcodeLogger`:
+
+The `opcodeLogger` tracer type provides detailed information about each opcode executed during the transaction. You can inspect the `StructLog` objects to understand the flow of execution in the EVM and identify any issues.
+
+##### Example Code:
+
+```javascript
+// Connect to the local node using Remix's web3 provider
+const web3 = new Web3(web3.currentProvider);
+
+// Function to trace a transaction using debug_traceTransaction with opcodeLogger
+async function traceTransaction(txHash) {
+   return web3.currentProvider.send({
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'debug_traceTransaction',
+      params: [txHash, { tracer: 'opcodeLogger', disableMemory: false, disableStack: false, disableStorage: false }]
+   });
+}
+
+// Example usage
+const txHash = '0x8fc90a6c3ee3001cdcbbb685b4fbe67b1fa2bec575b15b0395fea5540d0901ae';
+const result = await traceTransaction(txHash);
+console.log(result);
+```
+
+##### Response Schema:
+
+##### OpcodeTrace
+
+The `OpcodeTrace` object represents the trace of a transaction execution in the EVM.
+
+It contains the following fields:
+
+- `gas`: **number** - total gas used during the execution.
+- `failed`: **boolean** - indicates whether the transaction failed.
+- `returnValue`: **string | null** - hex-encoded return value of the transaction.
+- `structLogs`: **StructLog[]** - list of `StructLog` objects representing the opcodes executed during the transaction.
```

Comment:
Do the values in each `StructLog` entry refer to the state before or after executing the entry's opcode?

For example, looking at  the `PUSH1`s opcodes it seems the state is **before** executing (`pc` points to self entry, not to the next one). But looking at the `REVERT` opcode it seems **after** given the `REVERT` already populates the `reason` field.

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/pull/2774#discussion_r1698714078

Reply:
Great, with real values looks more clear.

Could we add a sentence mentioning the state displayed for each opcode refers to the state after executing said opcode?